Sources/OTel/Logging/Processing/Batch/OTelBatchLogRecordProcessor.swift

```diff
@@ -141,12 +141,7 @@ public actor OTelBatchLogRecordProcessor<Exporter: OTelLogRecordExporter, Clock:
             try await exporter.export(batch)
         } catch is CancellationError {
             // No-op
-        } catch {
-            logger.error("Failed to export log batch.", metadata: [
-                "error": "\(String(describing: type(of: error)))",
-                "error_description": "\(error)",
-            ])
-        }
+        } catch { }
```

Comment:
Could you file an issue to track adding it back? Otherwise I'm not sure how we'll debug exporting failures.

Reply:
Why would we want to add it back? I thought we agreed that the exporter itself is responsible for logging/recovering failures and therefore logging it additionally inside the processor would be repetitive.

Reply:
Should we make the export method non-throwing then, to reflect that in the type system?

Reply:
I'd keep it throwing so that folks writing their own processors can still deal with the exporter errors, just ignore it in the batching processor.

Reply:
I wonder if we just flip it - and make it the responsibility of the caller of the exporter's throwing function to handle the errors? In this case, the processor would log the error. The exporter itself wouldn't, because it's already throwing it.

Or was there a reason we didn't want the processor to have this log line?

Reply:
If we do add it back, I think it should be at `debug` level, not `error`: https://www.swift.org/documentation/server/guides/libraries/log-levels.html

Reply:
Hmm, good point. In my experience, logging right where the actual error occurs usually gives you the most details, resulting in more insightful error logs. For example, in a gRPC exporter we know exactly what the provided exporter configuration was and can include parts of it in error logs where useful, but a processor wouldn't even know about gRPC.

Reply:
I imagine the exporter would attach useful metadata on the thrown error, rather than just logs. But it might be helpful to do both.

I'm really not a fan of the `catch {}`, somehow feels wrong, because alternative exporter implementations might not log and then errors will get completely thrown away.

Reply:
How would that metadata inside the caught error make it into logger metadata though? The only thing we'd be able to do is use the error's `description` which may contain that useful metadata but wouldn't be structured logging anymore. I get your point about the empty `catch`, though perhaps that's just a matter of documenting that part of creating an exporter is to log errors.

Reply:
Right, that's fair. Maybe we define an error type that holds structured metadata, easy to merge into the logger's metadata?

Is the exporter always expected to be called by a processor? If so, I'm still wondering if we can just then make the method non-throwing. I suspect folks writing custom processors/exporters will be faced with the same question we are here.

Making it non-throwing would force the exporter to log its errors, and we'd get consistent behavior.

I'm also happy to punt on this question and reopen later. Definitely not a blocker, just want to make sure we don't forget about it.

Reply:
> Is the exporter always expected to be called by a processor?

Yes. There's no exporter without a processor. I see your point and will open an issue to track it separately. For the context of this PR, let's go with the empty `catch` statement inside the processor.